Gems/LmbrCentral/Code/Tests/Builders/LuaBuilderTests.cpp

```diff
@@ -62,6 +62,59 @@ namespace UnitTest
         AZ::ComponentApplication::Descriptor m_descriptor;
     };
 
+    TEST_F(LuaBuilderTests, ParseLuaScript_DependencyRegexStress_Success)
+    {
+        LuaBuilderWorker worker;
+
+        AssetBuilderSDK::ProductPathDependencySet pathDependencies;
+
+        char resolvedPath[AZ_MAX_PATH_LEN];
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath("@gemroot:LmbrCentral@/Code/Tests/Lua/dependencyRegexStress.lua", resolvedPath, AZ_MAX_PATH_LEN);
```

Comment:
There is a ResolvePath overload that results a FixedMaxPath.

There is no need to ever use AZ_MAX_PATH_LEN again.

```suggestion
        AZ::IO::FixedMaxPath resolvedPath;
        AZ::IO::FileIOBase::GetInstance()->ResolvePath(resolvedPath, "@gemroot:LmbrCentral@/Code/Tests/Lua/dependencyRegexStress.lua");
```